source/opt/local_access_chain_convert_pass.h

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2016 The Khronos Group Inc.
+// Copyright (c) 2016 Valve Corporation
+// Copyright (c) 2016 LunarG Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIBSPIRV_OPT_SSAMEM_PASS_H_
+#define LIBSPIRV_OPT_SSAMEM_PASS_H_
+
+
+#include <unordered_map>
+#include <unordered_set>
+#include <map>
+#include <algorithm>
+#include <utility>
+#include <queue>
+
+#include "def_use_manager.h"
+#include "module.h"
+#include "basic_block.h"
+#include "pass.h"
+
+namespace spvtools {
+namespace opt {
+
+// See optimizer.hpp for documentation.
+class LocalAccessChainConvertPass : public Pass {
+ public:
+  LocalAccessChainConvertPass();
+  const char* name() const override { return "sroa"; }
+  Status Process(ir::Module*) override;
+
+ private:
+   // Module this pass is processing
+   ir::Module* module_;
+
+   // Def-Uses for the module we are processing
+   std::unique_ptr<analysis::DefUseManager> def_use_mgr_;
+
+  // Map from function's result id to function
+  std::unordered_map<uint32_t, ir::Function*> id2function_;
+
+  // Set of verified target types
```

Comment:
Good questions.

As you likely have come to realize, the target type is documented at IsTargetType.

I have now moved the data member definitions below the function members so any readers will come across the "definition" of the term target type before any "use" of it. Hopefully that will reduce such questions.

"Verified" here means that they are variables that IsTargetVar() and other member functions have made a determination on and have been cached for future queries. I have changed the comments to make this a little clearer.